src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -773,49 +787,321 @@ private void cancelElectionsForReferenceIds(List<String> referenceIds) {
+  }
+
+  /** Helper class to hold classification results for DACs, users, and datasets. */
+  private static class DacUserClassification {
+    Set<Dac> autoOpenDacs = new HashSet<>();
+    Set<Dac> manualOpenDacs = new HashSet<>();
+    Set<User> autoOpenUsers = new HashSet<>();
+    Set<User> manualOpenUsers = new HashSet<>();
+    Set<Dataset> autoOpenDatasets = new HashSet<>();
+    Set<Dataset> manualOpenDatasets = new HashSet<>();
+    Map<Integer, Integer> autoOpenUserIds = new HashMap<>();
+  }
+
+  /** Classifies DACs and users based on automation rules. */
+  private DacUserClassification classifyDacsAndUsers(
+      Collection<Dac> dacsForDar, List<Dataset> datasetsForDar, List<User> adminAndChairUsers) {
+
+    Set<Integer> dacIds =
+        datasetsForDar.stream().map(Dataset::getDacId).collect(Collectors.toSet());
+
+    DacUserClassification result = new DacUserClassification();
+
+    for (Integer dacId : dacIds) {
+      boolean autoOpen = hasAutoOpenRule(dacId);
+      Integer autoOpenUserId = getAutoOpenRuleUserId(dacId);
+
+      Dac dac = findDac(dacsForDar, dacId);
+      Dataset dataset = findDataset(datasetsForDar, dacId);
+
+      if (autoOpen) {
+        addAutoOpen(result, dacId, autoOpenUserId, dac, dataset);
+        addUsers(result.autoOpenUsers, dacId, adminAndChairUsers, true);
+      } else {
+        addManualOpen(result, dac, dataset);
+        addUsers(result.manualOpenUsers, dacId, adminAndChairUsers, false);
       }
-      // If the dar is not a progress report, use the DAR template else use the PR template.
-      if (dar.getProgressReport()) {
-        // Use the reference ID to link the fact that this progress report will have been noted.
-        // the DAR Code at this point will be ambiguous.
-        emailService.sendNewProgressReportRequestEmail(
-            user, dacDatasetMap, researcherName, collection.getDarCode(), dar.getReferenceId());
+    }
+
+    return result;
+  }
+
+  /** Finds a DAC by its ID from a collection of DACs. */
+  private Dac findDac(Collection<Dac> dacs, Integer dacId) {
+    return dacs.stream().filter(d -> d.getDacId().equals(dacId)).findFirst().orElse(null);
+  }
+
+  /** Finds a dataset by its DAC ID from a list of datasets. */
+  private Dataset findDataset(List<Dataset> datasets, Integer dacId) {
+    return datasets.stream().filter(ds -> ds.getDacId().equals(dacId)).findFirst().orElse(null);
+  }
+
+  /** Checks if an auto-open rule exists for a given DAC ID. */
+  private boolean hasAutoOpenRule(Integer dacId) {
+    return dacAutomationRuleService.findAllByDacId(dacId).stream()
+        .anyMatch(
+            r ->
+                r.ruleType() == DACAutomationRuleType.AUTO_OPEN_DAR_FOR_ALL_MEMBERS
+                    && r.enabledByUserId() != null);
+  }
+
+  /** Retrieves the user ID associated with the auto-open rule for a given DAC ID. */
+  private Integer getAutoOpenRuleUserId(Integer dacId) {
```

Comment:
It feels like the method name should be more explicit about the value being returned.

```suggestion
  private Integer getAutoOpenRuleEnabledByUserId(Integer dacId) {
```